gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/FidoFactorUtils.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gravitee.am.gateway.handler.root.resources.endpoint;
+
+import io.gravitee.am.common.factor.FactorType;
+import io.gravitee.am.gateway.handler.common.factor.FactorManager;
+import io.gravitee.am.gateway.handler.root.service.user.UserService;
+import io.gravitee.am.identityprovider.api.DefaultUser;
+import io.gravitee.am.model.Factor;
+import io.gravitee.am.model.User;
+import io.gravitee.am.model.factor.EnrolledFactor;
+import io.gravitee.am.model.factor.EnrolledFactorSecurity;
+import io.gravitee.am.model.oidc.Client;
+import io.reactivex.Completable;
+import io.vertx.reactivex.ext.web.RoutingContext;
+
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import static io.gravitee.am.common.factor.FactorSecurityType.WEBAUTHN_CREDENTIAL;
+import static io.gravitee.am.common.factor.FactorType.FIDO2;
+import static io.gravitee.am.common.utils.ConstantKeys.ENROLLED_FACTOR_ID_KEY;
+import static io.gravitee.am.model.factor.FactorStatus.ACTIVATED;
+
+/**
+ * @author Ashraful Hasan (ashraful.hasan at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class FidoFactorUtils {
```

Comment:
Not sure about this class ... 

- Utils class should be static with only static methods
- I don't think the package is correct, should be more in a `service` or `utils` package rather than `endpoint`